### PR TITLE
[JS/WebGPU] support Concat.int32 operator

### PR DESCRIPTION
**Pull request by xhcao**

### CODE REVIEW

**Comment by xhcao** [comment]
@qjia7 @gyagp PTAL

File: onnxruntime/core/providers/js/operators/concat.cc [hunk 1]
```diff
@@ -6,41 +6,42 @@
 namespace onnxruntime {
 namespace js {
 
-ONNX_OPERATOR_VERSIONED_KERNEL_EX(
-    Concat,
-    kOnnxDomain,
-    1, 3,
-    kJsExecutionProvider,
-    (*KernelDefBuilder::Create())
-        .TypeConstraint("T", DataTypeImpl::GetTensorType<float>()),
```
Comment (qjia7) [inline onnxruntime/core/providers/js/operators/concat.cc:15]: Please reference here https://github.com/microsoft/onnxruntime/pull/16901/files#diff-a8e22b7e3708f7ae216482ad10f0bbc8e61ce562448bb299b4cf199d8b557251R15.
It seems like you can directly add the `int32_t` type as below
```
.TypeConstraint("T", DataTypeImpl::GetTensorType<float>()).TypeConstraint("T", DataTypeImpl::GetTensorType<int32_t>())

Comment (satyajandhyala) [reply onnxruntime/core/providers/js/operators/concat.cc:15]: @qjia7 Please don't pick up from the PRs under review. Look at the code base. @fs-eire replied on that PR that it does not work. Data type cannot be both float and in32_t at the same time.

Comment (qjia7) [reply onnxruntime/core/providers/js/operators/concat.cc:15]: @satyajandhyala Thanks for pointing it out. My original meaning maybe like this one https://github.com/microsoft/onnxruntime/pull/16952/files#diff-d8d6d238bdce4bb5db019a9aec28bbedc6dfcce10f5d522268e2f05be83753f5R27. I just thought we can register different types together.

**Comment by guschmue** [comment]
lint is nagging:
Error: Following source files are not formatted: (did you run "npm run format"?)
js/web/lib/wasm/jsep/webgpu/ops/concat.ts

**Comment by xhcao** [comment]
> lint is nagging: Error: Following source files are not formatted: (did you run "npm run format"?) js/web/lib/wasm/jsep/webgpu/ops/concat.ts

Sorry for this error, I used `npm run lint` and did not throw any warning or error, and thought it had also checked the format.

**Comment by xhcao** [comment]
Please take a look, thanks.

**Review by qjia7 — APPROVED** [review]
LGTM

@guschmue @fs-eire Please take a look, thanks!

**Comment by xhcao** [comment]
@fs-eire @guschmue @satyajandhyala Viewed the bots failures, it seemed that it was not related to the PR. Please trigger the bots to whether it could pass now.

**Review by fs-eire — APPROVED** [review]

**Pull request merged**